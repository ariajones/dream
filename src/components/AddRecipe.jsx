import React, { useState } from 'react'

import firebase from 'firebase/app'

import { addRecipe } from '../utils'

// import MultipleIngredientTest from './MultipleIngredientTest'

const AddRecipe = () => {
  const [name, setName] = useState('')
  const [imgPath, setImgPath] = useState('')
  const [serves, setServes] = useState('')
  const [prepTime, setPrepTime] = useState('')
  const [ingredients, setIngredients] = useState([])
  const [method, setMethod] = useState([])
  const [newIngredient, setNewIngredient] = useState('')

  function onSubmitHandler(e) {
    e.preventDefault()

    const newRecipe = {
      name,
      imgPath,
      serves,
      prepTime,
      ingredients,
      method
    }

    addRecipe(newRecipe)
    setName('')
    setImgPath('')
    setServes('')
    setPrepTime('')
    setIngredients([])
    setMethod([])
  }

  return (
    <>
    <form onSubmit={onSubmitHandler}>
      <h4>Add Recipe</h4>
      <div>
        <label>Title</label>
        <input type='text' value={name} onChange={e => setName(e.currentTarget.value)} />
      </div>
      <div>
        <label>Image URL</label>
        <input type='text' value={imgPath} onChange={e => setImgPath(e.currentTarget.value)} />
      </div>
      <div>
        <label>Serves</label>
        <input type='number' value={serves} onChange={e => setServes(e.currentTarget.value)} />
      </div>
      <div>
        <label>Prep Time</label>
        <input type='text' value={prepTime} onChange={e => setPrepTime(e.currentTarget.value)} />
      </div>
      <div>
        <label>{`Ingredients (seperate by comma)`}</label>
        <input type='text' value={ingredients} onChange={e => setIngredients(e.currentTarget.value.split(','))} />
        <ul>{ingredients.map(ingredient => <li key={ingredient}>{ingredient}</li>)}</ul>
      </div>
      <div>
        <label>{`Method (seperate by comma)`}</label>
        <input type='text' value={method} onChange={e => setMethod(e.currentTarget.value.split(','))} />
        <ol>{method.map(step => <li key={step}>{step}</li>)}</ol>
      </div>
      <button>Add Recipe</button>
    </form>

    {/* <MultipleIngredientTest /> */}
    </>
  )
}

export default AddRecipe;
