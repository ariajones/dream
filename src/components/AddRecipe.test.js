import React from 'react'
import renderer from 'react-test-renderer'

import AddRecipe from './AddRecipe'

it ('renders component correctly', () => {
    const blah = renderer
        .create(<AddRecipe/>)
        .toJSON()
    expect(blah).toMatchSnapshot()
})