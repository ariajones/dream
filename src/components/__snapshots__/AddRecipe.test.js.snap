// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h4>
    Add Recipe
  </h4>
  <div
    className="card"
  >
    <div
      className="card-image"
    >
      <figure
        className="image is-2by1"
      >
        <img
          alt="add your meal"
          src="https://icon-library.com/images/photo-placeholder-icon/photo-placeholder-icon-6.jpg"
        />
        <button
          className="button is-link is-small is-rounded"
        >
          <i
            className="fas fa-plus"
          />
        </button>
      </figure>
    </div>
    <div
      className="card-content"
    >
      <div
        className="media"
      >
        <div
          className="media-left"
        >
          <figure
            className="image is-48x48"
          />
        </div>
        <div
          className="media-content"
        >
          <p
            className="title is-5"
          >
            <div>
              <label>
                Title
              </label>
              <input
                className="label is-primary"
                onChange={[Function]}
                type="text"
                value=""
              />
               
            </div>
          </p>
        </div>
      </div>
    </div>
    <div>
      <label>
        Image URL
      </label>
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div
      className="content"
    >
      <div
        className="card-content"
      >
        Serves
        <br />
        <input
          onChange={[Function]}
          type="number"
          value=""
        />
         
      </div>
      <div
        className="card-content"
      >
        Prep Time
        <br />
        <input
          onChange={[Function]}
          type="text"
          value=""
        />
         
      </div>
    </div>
    <div
      className="card-content"
    >
      Ingredients needed: (seperate by comma)
      <div>
        <textarea
          className="textarea is-primary"
          onChange={[Function]}
          type="text"
          value={Array []}
        />
         
        <ul />
      </div>
    </div>
    <div
      className="card-content"
    >
      <div>
        <label>
          Method (seperate by comma)
        </label>
        <textarea
          className="textarea is-primary"
          onChange={[Function]}
          type="text"
          value={Array []}
        />
        <ol />
      </div>
      <br />
      <button
        className="button card-content is-medium is-rounded is-primary"
      >
        Add Recipe
      </button>
    </div>
  </div>
  <br />
</form>
`;
